solc/CommandLineInterface.cpp

```diff
@@ -1231,6 +1241,25 @@ void CommandLineInterface::outputCompilationResults()
 		handleNatspec(true, contract);
 		handleNatspec(false, contract);
 	} // end of contracts iteration
+
+	if (!g_hasOutputed &&
+			!m_args.count(g_argAst) &&
```

Comment:
Sorry, I think I asked this before, but why can't these be removed here?

Reply:
If the option -o is set, there may be no print on the screen but in the files, e.g. --ast.

Reply:
Ok, but then the message should be different, shouldn't it? So if `-o` was given, we could report something like `Generated output files in <directory>.`.

Reply:
I mean the point of doing it like this (i.e. with the global variable) was that we can add another commandline option without checking this condition here.

Reply:
updated according to your comments :-)